tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -398,6 +398,96 @@ def test_slurm_config_update(
     )
 
 
+@pytest.mark.usefixtures("region", "os", "instance", "scheduler")
+@pytest.mark.slurm_custom_config_parameters
+def test_slurm_custom_config_parameters(
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    test_datadir,
+    s3_bucket_factory,
+    scheduler_commands_factory,
+):
+    """Test slurm custom settings."""
+    # When launching a cluster with a Yaml with custom config parameters
+
+    # Prepare bucket
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "custom_slurm_settings.txt"), "custom_slurm_settings.conf")
+    slurm_settings_file = f" s3://{bucket_name}/custom_slurm_settings.conf"
+
+    cluster_config = pcluster_config_reader(bucket=bucket_name)
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    # then we expect custom values to be set in slurm config
+    # Bulk Settings
+    debug_flags = slurm_commands.get_conf_param("DebugFlags")
+    # must check them individually because they can be reordered
+    assert "Steps" in debug_flags
+    assert "Power" in debug_flags
+    assert "CpuFrequency" in debug_flags
+    assert "medium" == slurm_commands.get_conf_param("GpuFreqDef")
+    assert "5000" == slurm_commands.get_conf_param("MaxStepCount")
+
+    # Partition
+    assert "5" == slurm_commands.get_partition_info("q1", "GraceTime")
+    assert "500" == slurm_commands.get_partition_info("q1", "MaxMemPerNode")
+
+    # ComputeResource 1
+    assert "50" == slurm_commands.get_node_attribute("q1-dy-cr1-1", "Weight")
+    assert "10000" == slurm_commands.get_node_attribute("q1-dy-cr1-1", "Port")
+    assert "2000" == slurm_commands.get_node_attribute("q1-dy-cr1-1", "Memory")
+
+    # ComputeResource 2
+    assert "150" == slurm_commands.get_node_attribute("q1-dy-cr2-1", "Weight")
+    assert "10010" == slurm_commands.get_node_attribute("q1-dy-cr2-1", "Port")
+    assert "2500" == slurm_commands.get_node_attribute("q1-dy-cr2-2", "Memory")
```

Comment:
Nitpick: why are we testing a different node here?

Reply:
Same as above.